File: PadelClub.Services/Database/Order.cs
using System;

namespace PadelClub.Services.Database
{
    public class Order
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string OrderNumber { get; set; } = string.Empty;
        public decimal TotalAmount { get; set; }
        public string Status { get; set; } = "Pending"; // Pending, Processing, Shipped, Delivered, Cancelled
        public string RecipientName { get; set; } = string.Empty;
        public string PhoneNumber { get; set; } = string.Empty;
        public string ShippingAddress { get; set; } = string.Empty;
        public string City { get; set; } = string.Empty;
        public string PostalCode { get; set; } = string.Empty;
        public string? Notes { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        public virtual User User { get; set; } = null!;
        public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
        public virtual Payment? Payment { get; set; }
        public decimal SubtotalAmount { get; set; }
        public decimal DiscountAmount { get; set; }
        public int? CouponId { get; set; }
        public string? CouponCode { get; set; }
        public DateTime? DeliveredAt { get; set; }
        public virtual ICollection<ReturnRequest> ReturnRequests { get; set; } = new List<ReturnRequest>();
    }
}

File: UI/padelclub_desktop/lib/features/orders/domain/entities/order.dart
class Order {
  const Order({
    required this.id,
    required this.userId,
    required this.orderNumber,
    required this.totalAmount,
    required this.status,
    required this.recipientName,
    required this.phoneNumber,
    required this.shippingAddress,
    required this.city,
    required this.postalCode,
    required this.createdAt,
    this.notes,
  });

  final int id;
  final int userId;
  final String orderNumber;
  final double totalAmount;
  final String status;
  final String recipientName;
  final String phoneNumber;
  final String shippingAddress;
  final String city;
  final String postalCode;
  final DateTime createdAt;
  final String? notes;
}
